File: 0020. Valid Parentheses/Solution.cpp
#include <string>
#include <stack>

using std::string, std::stack;

class Solution {
public:
    bool isValid(string s) {
        stack<char> stack;
        for (char i:s)
        {
            if (i == '(' || i == '{' || i == '[') stack.push(i);
            else 
            {
                if (stack.empty() || (stack.top() == '(' && i != ')') || (stack.top() == '{' && i != '}') || (stack.top() == '[' && i != ']')) return false;
                stack.pop();
            }
        }
        return stack.empty();
    }
};

File: 0020. Valid Parentheses/Solution.py
class Solution:
    def isValid(self, s):
        """
        type s: str
        rtype: bool
        """
        stack = []
        for x in s:
            if x == '(' or x == '[' or x == "{": 
                stack.append(x)
            else:
                if len(stack) == 0 or (stack[-1] == "(" and x != ")") or (stack[-1] == "{" and x != "}") or (stack[-1] == "[" and x != "]"): 
                    return False
                stack.pop()
        return len(stack) == 0